### PR TITLE
OCPBUGS#30133: Updates to dynamic plugin docs

### PR DESCRIPTION
**Pull request by opayne1**
Updates to dynamic plugin docs

Version(s):
4.15+

Issue:
https://issues.redhat.com/browse/OCPBUGS-30133

Link to docs preview:
https://72479--ocpdocs-pr.netlify.app/openshift-enterprise/latest/web_console/dynamic-plugin/overview-dynamic-plugin

QE review:
- [x] QE has approved this change.

Additional information:
Provides doc updates for https://github.com/openshift/console/pull/13637

### CODE REVIEW

File: web_console/dynamic-plugin/overview-dynamic-plugin.adoc [hunk 1]
```diff
@@ -9,7 +9,7 @@ toc::[]
 [id="dynamic-plug-in-overview"]
 == About dynamic plugins
 
-A dynamic plugin allows you to add custom pages and other extensions to your interface at runtime. The `ConsolePlugin` custom resource registers plugins with the console, and a cluster administrator enables plugins in the `console-operator` configuration.
+Dynamic plugins are deployed as workloads on the cluster. They you to add custom pages and other extensions to your interface at runtime. The `ConsolePlugin` custom resource registers plugins with the console, and a cluster administrator enables plugins in the `console-operator` configuration.
```
Comment (vojtechszocs) [inline web_console/dynamic-plugin/overview-dynamic-plugin.adoc:12]:
```suggestion
Dynamic plugins are deployed as workloads on the cluster. They allow you to add custom pages and perform other extensions to console user interface at runtime. The `ConsolePlugin` custom resource registers plugins with the console, and a cluster administrator enables plugins in the `console-operator` configuration.
```

File: web_console/dynamic-plugin/overview-dynamic-plugin.adoc [hunk 2]
```diff
@@ -39,11 +39,15 @@ conster Header: React.FC = () => {
...
 * Avoid selectors that could affect markup outside of your plugins components, such as element selectors. These are not APIs and are subject to change. Using them might break your plugin. Avoid selectors like element selectors that could affect markup outside of your plugins components.
+* Provide valid JavaScript Multipurpose Inernet Mail Extension (MIME) type using the `Content-Type` response header for all assets served by your plugin web server. Each plugin deployment should include a web server that hosts the generated assets of the given plugin.
 
 [discrete]
 == PatternFly guidelines
 When creating your plugin, follow these guidelines for using PatternFly:
 
 * Use link:https://www.patternfly.org/components/all-components/[PatternFly] components and PatternFly CSS variables. Core PatternFly components are available through the SDK. Using PatternFly components and variables help your plugin look consistent in future console versions.
+** Use Patternfly 4.x if you are using {product-title} versions 4.14 and before.
+** Use Patternfly 5.x if you are using {product-title} 4.15 and later.
```
Comment (vojtechszocs) [inline web_console/dynamic-plugin/overview-dynamic-plugin.adoc:50]: :+1:

**Review by vojtechszocs — COMMENTED** [review]
@opayne1 Just a small suggestion but feel free to change it as necessary.

**Comment by opayne1** [comment]
@yapei could you help with a QE review of this PR? Thanks! 😄

**Comment by yapei** [comment]
looks good to me

File: web_console/dynamic-plugin/overview-dynamic-plugin.adoc [hunk 3]
```diff
@@ -39,11 +39,15 @@ conster Header: React.FC = () => {
 ----
 
 * Avoid selectors that could affect markup outside of your plugins components, such as element selectors. These are not APIs and are subject to change. Using them might break your plugin. Avoid selectors like element selectors that could affect markup outside of your plugins components.
+* Provide valid JavaScript Multipurpose Inernet Mail Extension (MIME) type using the `Content-Type` response header for all assets served by your plugin web server. Each plugin deployment should include a web server that hosts the generated assets of the given plugin.
```
Comment (gwynnemonahan) [inline web_console/dynamic-plugin/overview-dynamic-plugin.adoc:42]: Should "Multipurpose Inernet Mail Extension" be "Multipurpose Internet Mail Extension"?

File: web_console/dynamic-plugin/overview-dynamic-plugin.adoc [hunk 4]
```diff
@@ -39,11 +39,15 @@ conster Header: React.FC = () => {
...
 
 [discrete]
 == PatternFly guidelines
 When creating your plugin, follow these guidelines for using PatternFly:
 
 * Use link:https://www.patternfly.org/components/all-components/[PatternFly] components and PatternFly CSS variables. Core PatternFly components are available through the SDK. Using PatternFly components and variables help your plugin look consistent in future console versions.
+** Use Patternfly 4.x if you are using {product-title} versions 4.14 and before.
```
Comment (gwynnemonahan) [inline web_console/dynamic-plugin/overview-dynamic-plugin.adoc:49]: Per IBM Style Guide, it should read "using {product-title} 4.14 and earlier."

See: https://www.ibm.com/docs/en/ibm-style?topic=references-product-versions#references-to-earlier-or-later-versions

File: web_console/dynamic-plugin/overview-dynamic-plugin.adoc (hunk 2 above)
Comment (gwynnemonahan) [inline web_console/dynamic-plugin/overview-dynamic-plugin.adoc:50]: Per IBM Style Guide, it should read "using {product-title} 4.15 or later."

See: https://www.ibm.com/docs/en/ibm-style?topic=references-product-versions#references-to-earlier-or-later-versions

Comment (opayne1) [reply web_console/dynamic-plugin/overview-dynamic-plugin.adoc:50]: Thanks @gwynnemonahan I could not for the life of me find where that was in IBM style. I had a feeling they were wrong lol

**Comment by gwynnemonahan** [comment]
Hey @opayne1 ,

Couple style things, but otherwise, LGTM.

Thanks!

**Pull request merged**